sycl-fusion/jit-compiler/lib/KernelFusion.cpp

```diff
@@ -108,10 +116,10 @@ FusionResult KernelFusion::fuseKernels(
   bool CachingEnabled = ConfigHelper::get<option::JITEnableCaching>();
   CacheKeyT CacheKey{TargetArch,
```

Comment:
Now that we do not depend on STL and cache is not part of the headers, we might be able to use `SmallVector` instead of `std::vector` in the key. Not in this PR, ofc.